python/lsst/sims/catUtils/mixins/snUniversalRules.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+
+__all__ = ['SNUniverse']
+
+class SNUniverse(object):
+    """
+    Mixin Class providing methods for distributing SN model parameters
+    conditioned  on the parameters of the hostgalaxy from catsim Catalogs
+
+    The base class must have the following attributes:
+    numobjs
+    suppressHighzSN
+    badvalues
+    midSurveyTime
+    suppressDimSN
+    mjdobs
+    maxTimeSNVisible
+
+    """
+
+    @property
+    def averageRate(self):
+        if not hasattr(self, '_averageRate'):
+            hundredyear = 100. * 365.
+            self._averageRate = hundredyear
+        return self._averageRate
+
+    @averageRate.setter
+    def averageRate(self, value):
+        self._averageRate = value
+        return self._averageRate
+            
+    @property
+    def midSurveyTime(self):
+        '''
+        The time at the middle of the survey: ie. at the 5 year period.
+
+
+        .. note: Changing this should not change the statistical
+        properties of the survey, but will change the exact SN we find.
+        '''
+        if not hasattr(self, '_midSurveyTime'):
+            midSurveyTime_default = 570000.0
+            self._midSurveyTime = midSurveyTime_default
+        return self._midSurveyTime
+
+    @midSurveyTime.setter
+    def midSurveyTime(self, mymidSurveyTime):
+        '''
+        set the value of suppressDimSN of the catalog 
```

Comment:
Does this docstring belong with a different setter?  You are defining a setter for midSurveyTime, but the docstring says you are setting suppressDimSN.